File: tests/cases/fib_with_argparse.py
#!/usr/bin/env python3

from argparse_dataclass import dataclass


@dataclass
class Options:
    v: bool = False
    n: int = 0


def fib(i: int) -> int:
    if i == 0 or i == 1:
        return 1
    return fib(i - 1) + fib(i - 2)


if __name__ == "__main__":
    args = Options.parse_args()
    if args.n == 0:
        args.n = 5
    print(fib(args.n))

File: pyjl/tests/tests_py2many-py2many/fib_with_argparse.jl
using DataClass
using argparse_dataclass: dataclass
@dataclass mutable struct Options
v::Bool
n::Int64
_initvars = [_init=true, _repr=true, _eq=true, _order=false, _unsafe_hash=false, _frozen=false]
end

function fib(i::Int64)::Int64
if i == 0 || i == 1
return 1
end
return (fib((i - 1)) + fib((i - 2)))
end

function main()
args = ::from_args()
if args.n == 0
args.n = 5
end
println(fib(args.n));
end

main()
